### ISSUE TITLE
graph: maximum flow algorithm needed

### ISSUE THREAD

**Issue by monomonedula** [gonum/gonum#1413]
### What are you trying to do?

I need a tool to compute maximum flow value for a flow network.

Currently there's no implementation of any of max flow algorithms in golang.

I am willing to contribute with Edmonds-Karp algorithm implementation

**Comment by kortschak** [comment]
Please follow the template.

The work that you have in #1412 looks way more complicated than it needs to be and should be in the already existing graph/flow package.

I would recommend that you give some of the background and it you are prepared to do the work, sketch out an API for the functionality before you submit any pull request.

**Comment by monomonedula** [comment]
@kortschak Well, the current flow package is dedicated to control flow graphs, which is quite different field of graph theory

**Comment by kortschak** [comment]
Yeah, it's arguable. An alternative is graph/network which does already have flow like analysis in it.
